File: U4/C#/U4P_19.cs
using System;

namespace MyApp // Note: actual namespace depends on the project name.
{
    internal class U4P_19
    {
        static void Main(string[] args)
        {
            int filas = 3;
            int columnas = 3;

            // Crear una matriz
            int[,] matriz = new int[filas, columnas];

            InicializarMatriz(matriz);

            // Imprimir la matriz
            Console.WriteLine("Matriz:");
            ImprimirMatriz(matriz);
            ContarParesEImpares(matriz);
            Console.ReadKey();
        }
        static void InicializarMatriz(int[,] matriz)
        {
            Random random = new Random();

            for (int i = 0; i < matriz.GetLength(0); i++)
            {
                for (int j = 0; j < matriz.GetLength(1); j++)
                {
                    matriz[i, j] = random.Next(1, 101); // Números aleatorios entre 1 y 100
                }
            }
        }

        // Método para imprimir la matriz
        static void ImprimirMatriz(int[,] matriz)
        {
            for (int i = 0; i < matriz.GetLength(0); i++)
            {
                for (int j = 0; j < matriz.GetLength(1); j++)
                {
                    Console.Write(matriz[i, j] + " ");
                }
                Console.WriteLine();
            }
        }

        // Método para contar la cantidad de números pares e impares en la matriz
        static void ContarParesEImpares(int[,] matriz)
        {
            int pares = 0;
            int impares = 0;

            for (int i = 0; i < matriz.GetLength(0); i++)
            {
                for (int j = 0; j < matriz.GetLength(1); j++)
                {
                    if (matriz[i, j] % 2 == 0)
                    {
                        pares++;
                    }
                    else
                    {
                        impares++;
                    }
                }
            }

            Console.WriteLine($"\nCantidad de números pares: {pares}");
            Console.WriteLine($"Cantidad de números impares: {impares}");
        }
    }
}  

File: U4/Java/U4P_19.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompan.u4p_19;
import java.util.Random;
/**
 *
 * @author alfre
 */
public class U4P_19 {

    public static void main(String[] args) {
        int filas = 3;
        int columnas = 3;

        // Crear una matriz
        int[][] matriz = new int[filas][columnas];

        inicializarMatriz(matriz);

        // Imprimir la matriz
        System.out.println("Matriz:");
        imprimirMatriz(matriz);
        contarParesEImpares(matriz);
    }

    static void inicializarMatriz(int[][] matriz) {
        Random random = new Random();

        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[i].length; j++) {
                matriz[i][j] = random.nextInt(100) + 1; // Números aleatorios entre 1 y 100
            }
        }
    }

    // Método para imprimir la matriz
    static void imprimirMatriz(int[][] matriz) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[i].length; j++) {
                System.out.print(matriz[i][j] + " ");
            }
            System.out.println();
        }
    }

    // Método para contar la cantidad de números pares e impares en la matriz
    static void contarParesEImpares(int[][] matriz) {
        int pares = 0;
        int impares = 0;

        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[i].length; j++) {
                if (matriz[i][j] % 2 == 0) {
                    pares++;
                } else {
                    impares++;
                }
            }
        }

        System.out.println("\nCantidad de números pares: " + pares);
        System.out.println("Cantidad de números impares: " + impares);
    }
}
